specs/language/resources.tex

```diff
@@ -0,0 +1,943 @@
+   void InterlockedMax(in uint pos, in int value)
+   void InterlockedMax(in uint pos, in uint value)
+   void InterlockedMax(in uint pos, in int value, out int original)
+   void InterlockedMax(in uint pos, in uint value, out uint original)
+
+   // 32-bit integer atomic compare/exchange operations
+   void InterlockedCompareStore(in uint pos, in uint compare, in uint value);
+   void InterlockedExchange(in uint pos, in uint value, out uint original);
+   void InterlockedCompareExchange(in uint pos, in uint compare, in uint value,
+                                    out uint original);
+
+   // 64-bit integer atomic arithmatic/bitwise operations
+   void InterlockedAdd64(in uint pos, in uint64_t value);
+   void InterlockedAdd64(in uint pos, in uint64_t value, out uint64_t original)
+   void InterlockedAnd64(in uint pos, in uint64_t value);
+   void InterlockedAnd64(in uint pos, in uint64_t value, out uint64_t original)
+   void InterlockedOr64(in uint pos, in uint64_t value);
+   void InterlockedOr64(in uint pos, in uint64_t value, out uint64_t original)
+   void InterlockedXor64(in uint pos, in uint64_t value);
+   void InterlockedXor64(in uint pos, in uint64_t value, out uint64_t original)
+
+   // 64-bit integer atomic comparison operations
+   void InterlockedMin64(in uint pos, in int64_t value)
+   void InterlockedMin64(in uint pos, in int64_t value, out int64_t original)
+   void InterlockedMin64(in uint pos, in uint64_t value)
+   void InterlockedMin64(in uint pos, in uint64_t value, out uint64_t original)
+   void InterlockedMax64(in uint pos, in int64_t value)
+   void InterlockedMax64(in uint pos, in int64_t value, out int64_t original)
+   void InterlockedMax64(in uint pos, in uint64_t value)
+   void InterlockedMax64(in uint pos, in uint64_t value, out uint64_t original)
+
+   // 64-bit integer atomic compare/exchange operations
+   void InterlockedCompareStore64(in uint pos, in uint64_t compare, in uint64_t value);
+   void InterlockedExchange64(in uint pos, in int64_t value, out int64_t original);
+   void InterlockedCompareExchange64(in uint pos, in uint64_t compare,
+                                      in uint64_t value, out int64_t original);
+
+   // 32-bit float atomic compare/exchange operations
+   void InterlockedCompareStoreFloatBitwise(in uint byteOffest,
+                                             in float compare, in float value);
+   void InterlockedExchangeFloat(in uint byteOffest, in float value,
+                                  out float original);
+   void InterlockedCompareExchangeFloatBitwise(in uint byteOffest,
+                                                in float compare,
+                                                in float value,
+                                                out float original);
+};
+\end{HLSL}
+
+\Sub{Constructors}{Resources.rwbabufs.ctrs}
+
+Since their contents are viewed as raw bytes, RWByteAddressBuffers are defined as a simple variable definition.
```

Comment:
Wonder if "without any type parameters" might be more specific than "simple variable definition".

Reply:
Replaced